sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/PathsHelper.java

```diff
@@ -271,6 +273,10 @@ public static String generatePath(ResourceType resourceType, String ownerOrResou
             return Paths.DATABASE_ACCOUNT_PATH_SEGMENT;
         } else if (resourceType == ResourceType.DatabaseAccount) {
             return Paths.DATABASE_ACCOUNT_PATH_SEGMENT + "/" + ownerOrResourceId;
+        } else if (resourceType == ResourceType.ClientEncryptionKey) {
+            ResourceId clientEncryptionKeyId = ResourceId.parse(ownerOrResourceId);
+            return Paths.DATABASES_PATH_SEGMENT + "/" + clientEncryptionKeyId.getDatabaseId().toString() + "/" +
+                Paths.CLIENT_ENCRYPTION_KEY_PATH_SEGMENT + "/" + clientEncryptionKeyId.getUserId().toString();
```

Comment:
>yId.g [](start = 83, length = 5)

getClientEncryptionKeyId() instead of getUserId()

Reply:
Thanks for noticing . Done , we are not coming in this method for client encryption key that's why it got unnoticed